code/mission/missionmessage.cpp

```diff
@@ -403,6 +427,16 @@ void handle_legacy_backup_message(MissionMessage& msg, SCP_string wing_name) {
 	strcpy(msg.name, backup);
 }
 
+int lookup_mood(SCP_string& name) {
```

Comment:
We should be sticking to making things like this name parameter const as often as we can

Reply:
This is somewhere where my C++ is a little lacking. `SCP_string const &` means "a reference to an SCP_string which we promise not to change" and potentially allows for compiler optimizations, right? What about declaring the function itself `const`? (To be clear, I'm asking from an educational perspective ^^ )

Reply:
Yes, basically.
``SCP_string const &`` gives you a reference to a string you _cannot_ change. The compiler will stop you from any things that would change the string, so not only allows it for more optimization, but also is simply a safeguard for you as the developer to not do things you shouldn't.
Declaring the function const in this context is not possible. You could declare the return type const (``const int lookup_mood``), but declaring the _function_ const (``int lookup_mood(...) const {``) is only possible if it's a member function of a class (which it isn't, as a global function), and will basically mean "disallow any modification of member variables", which allows you to call the function on an object of that class which is const itself.